moto/codebuild/models.py

```diff
@@ -0,0 +1,265 @@
+
+        self.build_metadata["environment"] = {
+            "type": "LINUX_CONTAINER",
+            "image": "aws/codebuild/amazonlinux2-x86_64-standard:3.0",
+            "computeType": "BUILD_GENERAL1_SMALL",
+            "environmentVariables": [],
+            "privilegedMode": False,
+            "imagePullCredentialsType": "CODEBUILD",
+        }
+
+        self.build_metadata["serviceRole"] = service_role
+
+        self.build_metadata["logs"] = {
+            "deepLink": "https://console.aws.amazon.com/cloudwatch/home?region=eu-west-2#logEvent:group=null;stream=null",
+            "cloudWatchLogsArn": "arn:aws:logs:eu-west-2:{0}:log-group:null:log-stream:null".format(
+                get_account_id()
+            ),
+            "cloudWatchLogs": {"status": "ENABLED"},
+            "s3Logs": {"status": "DISABLED", "encryptionDisabled": False},
+        }
+
+        self.build_metadata["timeoutInMinutes"] = 45
+        self.build_metadata["queuedTimeoutInMinutes"] = 480
+        self.build_metadata["buildComplete"] = False
+        self.build_metadata["initiator"] = "rootme"
+        self.build_metadata[
+            "encryptionKey"
+        ] = "arn:aws:kms:eu-west-2:{0}:alias/aws/s3".format(get_account_id())
+
+
+class CodeBuild(BaseModel):
+    def __init__(
+        self,
+        region,
+        project_name,
+        project_source,
+        artifacts,
+        environment,
+        serviceRole="some_role",
+    ):
+        current_date = iso_8601_datetime_with_milliseconds(datetime.datetime.utcnow())
+        self.project_metadata = dict()
+
+        self.project_metadata["name"] = project_name
+        self.project_metadata["arn"] = "arn:aws:codebuild:{0}:{1}:project/{2}".format(
+            region, get_account_id(), self.project_metadata["name"]
+        )
+        self.project_metadata[
+            "encryptionKey"
+        ] = "arn:aws:kms:{0}:{1}:alias/aws/s3".format(region, get_account_id())
+        self.project_metadata[
+            "serviceRole"
+        ] = "arn:aws:iam::{0}:role/service-role/{1}".format(
+            get_account_id(), serviceRole
+        )
+        self.project_metadata["lastModifiedDate"] = current_date
+        self.project_metadata["created"] = current_date
+        self.project_metadata["badge"] = dict()
+        self.project_metadata["badge"][
+            "badgeEnabled"
+        ] = False  # this false needs to be a json false not a python false
+        self.project_metadata["environment"] = environment
+        self.project_metadata["artifacts"] = artifacts
+        self.project_metadata["source"] = project_source
+        self.project_metadata["cache"] = dict()
+        self.project_metadata["cache"]["type"] = "NO_CACHE"
+        self.project_metadata["timeoutInMinutes"] = ""
+        self.project_metadata["queuedTimeoutInMinutes"] = ""
+
+
+class CodeBuildBackend(BaseBackend):
+    """find which functions I used in the dev branch stuff"""
```

Comment:
Just as an FYI: Docstrings like this will end up on `docs.getmoto.org` as an explanation, so they can be useful if you want to explain what is/isn't supported. This comment looks more like a reminder to yourself though :)